src/app/query/bitcoin/ordinals/use-inscriptions.query.ts

```diff
@@ -48,18 +54,26 @@ export function useInscriptionsInfiniteQuery() {
   const network = useCurrentNetwork();
   const keychain = useTaprootCurrentAccountPrivateKeychain();
   const currentAccountIndex = useCurrentAccountIndex();
+  // we need to save the current account index to reset the account data when the account changes
+  const savedCurrentAccountIndex = useRef(currentAccountIndex);
 
   // TO-DO remove code for testing purposes before merge
   const searchParams = new URLSearchParams(document.location.search);
   const testInscriptionAddress = searchParams.get('testAddress');
-  const currentNumberOfAddressesWithoutOrdinals = useRef(0);
-
-  const fromIndex = useRef(0);
-  const addressesMap = useRef(
-    getAddressesData(fromIndex.current, fromIndex.current + addressesSimultaneousFetchLimit)
-  );
-  const queryParams = useRef(formQueryParams(addressesMap.current));
 
+  const defaultAccountData = {
+    fromIndex: 0,
+    addressesWithoutOrdinalsNum: 0,
+    addressesMap: getAddressesData(0, addressesSimultaneousFetchLimit),
+  };
+  const currentAccData = useRef<AccountData>(defaultAccountData);
+  const queryParams = useRef(formQueryParams(currentAccData.current.addressesMap));
```

Comment:
Just a note, in my PR I really needed to use the fetcher for a single address, but was a funky API because it expects a pre-formed query param. Why not just save the array of addresses, pass them in in the fetch fn, and make the url params there?

Reply:
good catch, thanks